### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by noelfoka**
## Bookstore: add reducers and actions

In this branch, I:

- [ ] Add Redux Toolkit (`npm install react-redux @reduxjs/toolkit`).
- [ ] Create a directory containing all your Redux logic (`/src/redux`).
- [ ] Configure a Redux store (`/src/redux/store.js`).
- [ ] Define a slice of state for books that:
    - Includes an array of books.
    - Includes a reducer that adds a book.
    - Includes a reducer that removes a book

- [ ] Define a `slice` of state for `categories` that:
    - Includes an array of categories.
    - Includes a reducer that checks the status and always returns "Under construction".

### CODE REVIEW

File: src/redux/store.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import booksReducer from './books';
+import categoryReducer from './categories';
+
+const store = configureStore({
+  reducer: {
+    books: booksReducer,
+    categories: categoryReducer,
+  },
+});
+
```
Comment (ademibowale) [inline src/redux/store.js]: - [ ] You have done a great job on this milestone but there are still some issues you still need to address, According to the project requirement kindly Structure your application files using a "feature folder" approach and use the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for your Redux files. Your files/folder structure could look like this:
- [ ] see project requirement screenshot below
22 08 2023_10 12 21_REC

File: src/redux/books.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      title: 'Dune',
+      author: 'Frank Herbert',
+      category: 'Science Fiction',
+    },
+    {
+      id: 3,
+      title: 'Capital in the Twenty-First Century',
+      author: 'Suzanne Collins',
+      category: 'Economy',
+    },
```
Comment (ademibowale) [inline src/redux/books.js]: - [ ] I will suggest you follow the project requirement kindly go through the requirement according to the project requirement and consider to Define a slice of state for books that:
Includes an array of books (initial state: empty array)
Includes a reducer that adds a book
Includes a reducer that removes a book

File: src/redux/categories.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  switch (action.type) {
+    case CHECK_STATUS:
+      return 'Under construction';
+    default:
+      return state;
+  }
+}
+
+export const checkStatus = () => ({
+  type: CHECK_STATUS,
```
Comment (ademibowale) [inline src/redux/categories.js]: - [ ] I will suggest you to follow the project requirement according to the requirement Define a slice of state for categories that:
Includes an array of categories (initial state: empty array)
Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)

**Review by ademibowale — CHANGES_REQUESTED** [review]
## STATUS: Changes Required :recycle:

### Hi @noelfoka  , 👋 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To Highlight 🥇

✅ Descriptive Pull Request title and summary. ✔️
✅ Good use of Github flow ✔️
✅All linters are checked.
✅Professional pull request title and description.

### Changes Required

- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @Adebowale ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by joskalenda — APPROVED** [review]
| STATUS: APPROVED ✅ |
|------------------------|
| Hi @noelfoka , 🥰|
| Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: Congratulations! 🎉  |
|  |

### Highlights
✅ ... Good redux actions
✅ ... Green linters

---------
### Optional suggestions

> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### Cheers and Happy coding!👏👏👏
-------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**